File: BinarySearch.cpp
#include <iostream>
using namespace std;
int BinarySearch(int arr[],int key,int size)
{
    int start=0;
    int End=size-1;
    int mid=((start+End)/2);
    while(start<=End)
    {
        if(arr[mid]==key)
        {
            return mid;
        }
        if(key>arr[mid])
        {
            start=mid+1;
        }
        else{
            End=mid-1;
        }
        mid=(start+End)/2;
    }
    return -1;
}
int main(){
    int arr[]={1,15,19,20,36};
    cout<<BinarySearch(arr,16,5);
    return 0;
}

File: BinarySearch.java
public class BinarySearch {
    public static void main(String[] args) {
        int[] arr=new int[]{1,2,3,4,5};
        System.out.println(binarySearching(arr,100,0,arr.length-1));
    }
    public static int binarySearching(int[] arr,int key,int start,int end)
    {
        if(start>end) return -1;
        int mid=(start+end)/2;
        if(arr[mid]==key) return mid;
        if(key>mid) return binarySearching(arr,key,mid+1,end);
        else return binarySearching(arr,key,start,mid-1);
    }
}
